doc/man3/STORE_LOADER.pod

```diff
@@ -0,0 +1,202 @@
+=pod
+
+=head1 NAME
+
+STORE_LOADER, STORE_LOADER_CTX, STORE_LOADER_new,
+STORE_LOADER_set0_scheme, STORE_LOADER_set_open,
+STORE_LOADER_set_ctrl, STORE_LOADER_set_load, STORE_LOADER_set_eof,
+STORE_LOADER_set_close, STORE_LOADER_free, STORE_register_loader,
+STORE_unregister_loader, STORE_open_fn, STORE_ctrl_fn, STORE_load_fn,
+STORE_eof_fn, STORE_close_fn - Types and functions to manipulate,
+register and unregister STORE loaders for different URI schemes 
+
+=head1 SYNOPSIS
+
+ #include <openssl/store.h>
+
+ typedef struct store_loader_st STORE_LOADER;
+
+ STORE_LOADER *STORE_LOADER_new(void);
+ int STORE_LOADER_set0_scheme(STORE_LOADER *store_loader, const char *scheme);
+
+ /* struct store_loader_ctx_st is defined differently by each loader */
+ typedef struct store_loader_ctx_st STORE_LOADER_CTX;
+
+ typedef STORE_LOADER_CTX *(*STORE_open_fn)(const char *uri,
+                                            const UI_METHOD *ui_method,
+                                            void *ui_data);
+ int STORE_LOADER_set_open(STORE_LOADER *store_loader,
+                           STORE_open_fn store_open_function);
+ typedef int (*STORE_ctrl_fn)(STORE_LOADER_CTX *ctx, int cmd, va_list args);
+ int STORE_LOADER_set_ctrl(STORE_LOADER *store_loader,
+                           STORE_ctrl_fn store_ctrl_function);
+ typedef STORE_INFO *(*STORE_load_fn)(STORE_LOADER_CTX *ctx,
+                                      UI_METHOD *ui_method, void *ui_data);
+ int STORE_LOADER_set_load(STORE_LOADER *store_loader,
+                           STORE_load_fn store_load_function);
+ typedef int (*STORE_eof_fn)(STORE_LOADER_CTX *ctx);
+ int STORE_LOADER_set_eof(STORE_LOADER *store_loader,
+                            STORE_eof_fn store_eof_function);
+ typedef int (*STORE_close_fn)(STORE_LOADER_CTX *ctx);
+ int STORE_LOADER_set_close(STORE_LOADER *store_loader,
+                            STORE_close_fn store_close_function);
+ void STORE_LOADER_free(STORE_LOADER *store_loader);
+ 
+ int STORE_register_loader(STORE_LOADER *loader);
+ STORE_LOADER *STORE_unregister_loader(const char *scheme);
+
+=head1 DESCRIPTION
+
+These functions help applications and engines to create loaders for
+schemes they support.
+
+=head2 Types
+
+B<STORE_LOADER> is the type to hold a loader.  It contains a scheme
+and the functions needed to implement STORE_open(), STORE_eof(),
+STORE_load() and STORE_close() for this scheme.
+
+B<STORE_LOADER_CTX> is a type template, to be defined by each loader
+using B<struct store_loader_ctx_st { ... }>.
```

Comment:
I'm not the one wanting to do something new and not getting an approval as yet so the motivation is stronger I would think for you to do that rather than me :-)